### ISSUE TITLE
The sample test is searching for libsodium.so.18, I have libsodium.18.dylib

### ISSUE THREAD

**Issue by bobgus** [FedericoCeratto/nim-libsodium#1]
I am using the sample test shown in the documentation.

`import libsodium.sodium`
`import libsodium.sodium_sizes`
`let`
`  msg = "hello and goodbye"`
`  (pk, sk) = crypto_box_keypair()`
`  nonce = randombytes(crypto_box_NONCEBYTES())`
`  ciphertext = crypto_box_easy(msg, nonce, pk, sk)`

Hmm, leading spaces get stripped..

I installed libsodium using `brew install libsodium`
This installation seemed to be a success.

On my Mac, I can see the pointers to the library:

air:Nim bobgus$ ls -l /usr/local/lib/libsodium.dylib
lrwxr-xr-x  1 bobgus  admin  46 Nov 30 15:00 /usr/local/lib/libsodium.dylib -> ../Cellar/libsodium/1.0.11/lib/libsodium.dylib

air:Nim bobgus$ ls -l /usr/local/Cellar/libsodium/1.0.11/lib/libsodium.dylib
lrwxr-xr-x  1 bobgus  admin  18 Nov 30 15:15 /usr/local/Cellar/libsodium/1.0.11/lib/libsodium.dylib -> libsodium.18.dylib

The error output is

> air:Nim bobgus$ nim c -r -d:nimDebugDlOpen libsodium_test.nim
Hint: used config file '/Users/bobgus/Nim/config/nim.cfg' [Conf]
libsodium_test.nim(8, 3) Hint: 'ciphertext' is declared but not used [XDeclaredButNotUsed]
CC: stdlib_system
Hint:  [Link]
Hint: operation successful (15188 lines compiled; 0.627 sec total; 21.504MiB; Debug Build) [SuccessX]
Hint: /Users/bobgus/Nim/libsodium_test  [Exec]
dlopen(libsodium.so.18, 2): image not found
could not load: libsodium.so.18

-----
I think I'm pretty close, but need some hints. Thanks much

**Comment by FedericoCeratto** [comment]
Thanks for the bug report @bobgus - can you please try now?

**Comment by bobgus** [comment]
I think sodium_sizes.nim needs a tweek too.

Federico Ceratto wrote:

**Comment by FedericoCeratto** [comment]
@bobgus as you can probably imagine I don't have a mac os host at hand and I'm relying on TravisCI. The imports are fixed but one of the tests is failing:  https://travis-ci.org/FedericoCeratto/nim-libsodium/builds/180682943  Can please test?

**Comment by bobgus** [comment]
Looks good - seems to get the correct library for macosx. The little test runs with no errors (and no output, but none was requested)

**Comment by bobgus** [comment]
I took a look at the travis-cl.org link and he is getting the exact same 
error.

The routine 'finalize' is having a problem. I put an echo statement in 
the code and it appears that the nil problem is occurring at the end of 
the calling list. No problem with finalize with earlier items in the list.

I commented out that whole test and everything else runs to completion.

Have fun - BobG

Federico Ceratto wrote:
